### ISSUE TITLE
100.64.0.0/10 is not included in allowed LAN ranges

### ISSUE THREAD

**Issue by nothingmuch** [mullvad/mullvadvpn-app#6086]
### Is it a bug?

- [X] I know this is an issue with the app, and contacting Mullvad support is not relevant.

### I have checked if others have reported this already

- [X] I have checked the issue tracker to see if others have reported similar issues.

### Current Behavior

Although 100.64.0.0/100 is part of the IPv4 reserved address blocks used for internal networks, it is not included in the list of allowed addresses when LAN access is unblocked.

Strictly speaking that range is for carrier-grade NAT, so it's debatable whether or not that should be considered LAN.

However, not adding this manually to to nftables means I can't use tailscale and mullvad concurrently.

I'm aware of tailscale's mullvad integration, but unfortunately since I use headscale and not the tailscale service, I can't yet utilize this functionality.

### Expected Behavior

With tailscale enabled and connected, `mullvad lan set allow`, when connecting to mullvad traffic on the tailscale interface should not be blocked.

I can work around this using the following firewall rule:

```
    table inet mullvadExclusions {
      chain mullvadAllowTailsacle {
        type route hook output priority -1; policy accept;
        ip daddr 100.64.0.0/10 ct mark set 0x00000f41 meta mark set 0x6d6f6c65;
      }
```

but it would be less brittle and more secure if I would not have to do that statically, since that would still be allowed if `mullvad lan set block` is configured.

### Steps to Reproduce

1. `tailscale up`
2. `mullvad lan set allow`
3. `tailscale ping ...` (works)
4. `mullvad connect`
5. `tailscale ping ...` (no response)
6. `mullvad disconnect`
7. `tailscale ping ...` (works)

### Operating system version

NixOS unstable (pre release 24.05)

### Mullvad VPN app version

2023.6

**Comment by harbor6688** [comment]
You could for now manually add it towards talpid-core/src/firewall/mod.rs in the sourcecode at https://github.com/mullvad/mullvadvpn-app/blob/6868d3c3336a8289c56fdd978ad61ef8a19a7155/talpid-core/src/firewall/mod.rs
 and modify ALLOWED_LAN_NETS
 while rust code can be a bit scary i feel that this part is pretty self explanatory.
and then manually compiling from said modified source preferably with build.sh --optimized so you get a release build now i cannot guarantee this to work since i have yet to test it myself but it could be a stopgap solution for now.
Now only issue with this would be that you would not run the release version but the latest git commit unless you can figure out what commit they used to compile the latest release version.

**Comment by harbor6688** [comment]
nvm i think this should be the commit of the latest release https://github.com/mullvad/mullvadvpn-app/commit/b261238598f0237aaf420354445797a12a45d907

**Comment by aktaboot** [comment]
I tried patching mullvad-vpn by adding in the subnet in ALLOWED_LAN_NETS but that does not seem to have any effect, which is weird to say the least.

**Comment by probablypablito** [comment]
Please add this range! Or even better allow custom ranges as WireGuard does :)

**Comment by avc123** [comment]
I too feel we should have the ability to exempt our own custom ranges.

**Comment by bak1an** [comment]
Having ability to exclude custom ranges would be great indeed.

**Comment by hossein1376** [comment]
For anyone facing the same issue, this [comment](https://github.com/mullvad/mullvadvpn-app/pull/6089#issuecomment-2315072330) hints at a solution. It only works on Linux.

Save the following config into a file (like `cfg`) and load it as root with `nft -f ./cfg`.

```nft
table inet excludeTraffic {
  chain excludeOutgoing {
    type route hook output priority -100; policy accept;
    ip daddr 100.64.0.0/10 ct mark set 0x00000f41 meta mark set 0x6d6f6c65
  }

  chain excludeIncoming {
    type filter hook input priority -100; policy accept;
    ip saddr 100.64.0.0/10 ct mark set 0x00000f41 meta mark set 0x6d6f6c65
  }
}
```

Disable MagicDNS with: `tailscale set --accept-dns=false`.

In the Mullvad app, there is no need to enable `Local network sharing`, but you have to enable the `Use custom DNS server`. First server must be `100.100.100.100`, and at least another server like `1.1.1.1` or similar is required.

**Comment by 0x09AF** [comment]
@hossein1376 nice solution, thank you. This works perfectly.  
The only thing I could not get working is shared-in accepted routes from the tailnet. I have the toggle "Local network sharing" turned on and the shared route is within the Class A network (10.0.0.0/8). Also, in `tailscale` the `--accept-routes` flag is set to `true`.  
Do you or anyone else know how to access accepted routes?

**Comment by rafa-br34** [comment]
This is a duplicate of #5921 
I found a way to use Tailscale as a SOCKS5 proxy on Windows. If anyone is interested, it can be found [here](https://github.com/mullvad/mullvadvpn-app/issues/5921#issuecomment-3557975087).

**Comment by azban** [comment]
@hossein1376 

> In the Mullvad app, there is no need to enable Local network sharing, but you have to enable the Use custom DNS server. First server must be 100.100.100.100, and at least another server like 1.1.1.1 or similar is required.

Does this still work for you? Why is 1.1.1.1 required? In case you aren't connected to tailscale? This is only resolving specific hostnames specified in magic dns (headscale), but isn't resolving node names or any other sites.

**Comment by hossein1376** [comment]
@azban  
> Does this still work for you?

Due to nationwide internet shutdown in Iran, I basically have had no internet for the past four months. Currently, I'm not able to verify it's still working.

> Why is 1.1.1.1 required? In case you aren't connected to tailscale? This is only resolving specific hostnames specified in magic dns (headscale), but isn't resolving node names or any other sites.

With MagicDNS disabled, `100.100.100.100` resolves tailscale specific routes. Another address (like `1.1.1.1`) is required to resolve any other routes.

**Comment by tritaum** [comment]
For anyone that needs an immediate solution for Linux I just created this mullvad daemon upstream-synced fork which allows you to exclude IPs/IP ranges from the mullvad tunnel, its very straightforward to setup, just replace your current daemon with the one in [Releases](https://github.com/tritaum/ip-split-tunneling-mullvad/releases) or install it directly using the AUR package named [patched-mullvad-vpn-daemon-bin](https://aur.archlinux.org/packages/patched-mullvad-vpn-daemon-bin)

If you want to check the source code, report bugs or make suggestions theres a Github repo for that:
https://github.com/tritaum/ip-split-tunneling-mullvad

If you're from another platform and need a solution for it you can create an issue in my repo
